pkg/webui/console/views/device-overview/index.js

```diff
@@ -67,6 +81,18 @@ class DeviceOverview extends React.Component {
     shouldRedirect: false,
   }
 
+  @bind
+  async onExport() {
+    const {
+      appId,
+      device: { ids, mac_settings },
+    } = this.props
+    const result = await tts.Applications.Devices.getById(appId, ids.device_id, ['mac_state'])
```

Comment:
Please add some error handling here. The request might fail or the response might not actually contain the `mac_state`. In such cases, the user should be notified with a standard error toast.

Reply:
Additionally, we can avoid making this request when the device has no current session:

```suggestion
    if (session) {
      const result = await tts.Applications.Devices.getById(appId, ids.device_id, ['mac_state'])
      …
```